medium-tests/src/test/java/mediumtest/synchronization/ConnectionSyncMediumTests.java

```diff
@@ -145,19 +148,71 @@ void it_should_notify_client_on_invalid_token_exactly_once() {
       .withResponseCodes(responseCodes -> responseCodes.withStatusCode(status))
       .start();
 
-    var backend = newBackend()
+    newBackend()
       .withSonarQubeConnection(CONNECTION_ID, server, storage -> storage.withPlugin(TestPlugin.JAVA).withProject("projectKey"))
       .withBoundConfigScope(SCOPE_ID, CONNECTION_ID, "projectKey")
       .withEnabledLanguageInStandaloneMode(JAVA)
       .withProjectSynchronization()
       .withFullSynchronization()
       .start(client);
+
     await().untilAsserted(() -> assertThat(client.getLogMessages()).contains("Error during synchronization"));
+    await().during(5, TimeUnit.SECONDS).untilAsserted(() -> assertThat(client.getConnectionIdsWithInvalidToken(CONNECTION_ID)).isEqualTo(1));
+  }
 
-    backend.getConnectionService().didChangeCredentials(new DidChangeCredentialsParams(CONNECTION_ID));
+  @SonarLintTest
+  void it_should_renotify_client_on_invalid_token_after_connection_is_recreated() {
+    var status = 401;
+    var client = newFakeClient()
+      .withCredentials(CONNECTION_ID, "user", "pw")
```

Comment:
I think there's a long term plan to remove user-password auth from SQC. Not sure about SQS, but can be as well. Maybe we should rely mostly on token auth in tests. Not urgent to fix now.

Reply:
Good point, I replaced with tokens in this class to make it easier later